src/main/java/com/thealgorithms/scheduling/EnergyAwareScheduling.java

```diff
@@ -0,0 +1,46 @@
+package com.thealgorithms.scheduling;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * EnergyAwareScheduling schedules tasks based on their energy consumption profile,
+ * aiming to minimize the overall energy usage of the system.
+ *
+ * Use Case: Mobile devices, embedded systems, and data centers that need to save power
+ * while maintaining performance.
+ *
+ * @author Hardvan
+ */
+public final class EnergyAwareScheduling {
+
+    static class Task {
+        String name;
+        int energyConsumption;
+
+        Task(String name, int energyConsumption) {
+            this.name = name;
+            this.energyConsumption = energyConsumption;
```

Comment:
Doesn't energy consumption depend on execution time? With the current implementation execution order makes no difference because the total energy consumption is always the same

Reply:
You are right, I have modified the code and test cases to reflect the same.